### PR TITLE
feat(google-rules-mvp): Hide axe color contrast when ATFA rules are enabled

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

Exclude Axe ColorContrast rules from NeedsReview tab if ATFA rules are enabled, as ATFA has its own color contrast rule that we believe to be less noisy. Also includes some strict null checks that were necessary to get `yarn fastpass` to work as expected. These are called out individually in the review.

##### Motivation

Requirement from the spec

#### Pull request checklist

- [x] Addresses an existing issue: https://mseng.visualstudio.com/1ES/_workitems/edit/1856549
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/platform/android/atfa-scan-results-to-unified-results.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function convertAtfaScanResultsToUnifiedResults(
             const viewElement: ViewHierarchyElement =
                 atfaResult['AccessibilityHierarchyCheckResult.element'];
             if (viewElement) {
-                const ruleInformation: RuleInformation = ruleInformationProvider.getRuleInformation(
```
Comment (DaveTryon) [inline src/electron/platform/android/atfa-scan-results-to-unified-results.ts:38]: Changed for strict null checks

File: src/electron/platform/android/axe-scan-results-to-unified-results.ts [hunk 2]
```diff
@@ -26,9 +26,7 @@ export function convertAxeScanResultsToUnifiedResults(
     const unifiedResults: UnifiedResult[] = [];
 
     for (const ruleResult of scanResults.ruleResults) {
-        const ruleInformation: RuleInformation = ruleInformationProvider.getRuleInformation(
-            ruleResult.ruleId,
-        );
+        const ruleInformation = ruleInformationProvider.getRuleInformation(ruleResult.ruleId);
```
Comment (DaveTryon) [inline src/electron/platform/android/axe-scan-results-to-unified-results.ts:29]: Changed for strict null checks

File: src/electron/platform/android/rule-information-provider-type.ts [hunk 3]
```diff
@@ -4,5 +4,5 @@
 import { RuleInformation } from './rule-information';
 
 export interface RuleInformationProviderType {
-    getRuleInformation(ruleId: string): RuleInformation;
+    getRuleInformation(ruleId: string): RuleInformation | null;
```
Comment (DaveTryon) [inline src/electron/platform/android/rule-information-provider-type.ts:7]: Changed for strict null checks

File: src/electron/platform/android/rule-information-provider.ts [hunk 4]
```diff
@@ -265,11 +271,25 @@ export class RuleInformationProvider {
         }
     };
 
-    public getRuleInformation(ruleId: string): RuleInformation {
+    public getRuleInformation(ruleId: string): RuleInformation | null {
```
Comment (DaveTryon) [inline src/electron/platform/android/rule-information-provider.ts:274]: Changed for strict null checks

File: src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts [hunk 5]
```diff
@@ -249,4 +260,22 @@ describe('RuleInformationProvider', () => {
             expect(ruleInformation.getResultStatus(ruleResult)).toBe(testCase.outcome);
         },
     );
+
+    it.each([
+        ['available when ATFA flag is disabled', false],
+        ['not available when ATFA flag is enabled', false],
```
Comment (lisli1) [inline src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts:266]: Should this be `true` for enabled feature flag?

**Review by lisli1 — APPROVED** [review]
LGTM, one comment inline

File: src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts (hunk 5 above)
Comment (DaveTryon) [reply src/tests/unit/tests/electron/platform/android/rule-information-provider.test.ts]: Durp! Fixed.

**Pull request merged**